src/ray/core_worker/core_worker.cc

```diff
@@ -1211,9 +1211,11 @@ Status CoreWorker::CancelTask(const ObjectID &object_id, bool force_kill) {
     return Status::Invalid("Actor task cancellation is not supported.");
   }
   rpc::Address obj_addr;
-  if (!reference_counter_->GetOwner(object_id, nullptr, &obj_addr) ||
-      obj_addr.SerializeAsString() != rpc_address_.SerializeAsString()) {
-    return Status::Invalid("Task is not locally submitted.");
+  if (!reference_counter_->GetOwner(object_id, nullptr, &obj_addr)) {
+    RAY_LOG(ERROR) << "No owner for object";
```

Comment:
`ERROR`-level warnings are be user-facing because they're printed to stderr, so please add more context to this error message.

Reply:
Shouldn't we also be returning a bad status here if we don't have the owner? That would be preferable to just logging, as it will be presented as an exception in python

Reply:
Oh, that would make a lot more sense.